docs(self-hosted configuration): improve username config option section (#14661)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

## username

> You might need to set a custom `username` if you use:

Reviewer: ```suggestion
You might need to set a `username` if you use:
```

- The Bitbucket platform, or
- use the GitHub App with CLI (required)

> If you're using a Personal Access Token (PAT) to authenticate then we recommend you keep the `username` at its default value.

Reviewer: ```suggestion
If you're using a Personal Access Token (PAT) to authenticate then you should not specify `username`.
```

<!-- prettier-ignore  -->
!!! caution

>     If you use a Personal Access Token (PAT) and want to set a custom `username` (not recommended!), then make sure that Renovate's `username` matches the PAT username.

Reviewer: I think remove all the above
